test/over_react/component_declaration/flux_component_test/component2/unsound_flux_component_test.dart

```diff
@@ -0,0 +1,85 @@
+//@dart=2.11
+
+// Copyright 2024 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Tests adapted from <https://github.com/Workiva/w_flux/blob/1.0.1/test/component_test.dart>.
+
+library over_react.component_declaration.component2.unsound_flux_component_test;
+
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_util/test_util.dart';
+import 'flux_component_test.dart';
+
+// These tests are only possible in unsound null safety, otherwise we'll get null errors
+// when the mocks return null when they shouldn't.
+void main() {
+  void sharedComponentTests(BaseTestComponents testComponents) {
+    group(
+        'mounts, rerenders, and unmounts properly when using a mock store that returns null'
+        ' for members like .listen and isOrWillBeDisposed', () {
+      test('basic usage', () async {
+        final mockStore = MockTestStore();
+        final jacket = mount((testComponents.basic()..store = mockStore)());
+        await pumpEventQueue();
+        verify(mockStore.listen(any));
```

Comment:
Should this also call `isOrWillBeDisposed` somewhere or is that already being tested under the hood?

Reply:
It should already be tested under the hood, but also verifying that wouldn't hurt. Added that and some more thorough comments here: https://github.com/Workiva/over_react/pull/885/commits/32f6d45bea6539d7c246bf89bf1e808672514e09